packages/iframe-stamper/src/index.ts

```diff
@@ -145,29 +149,29 @@ export class IframeStamper {
 
     // This is populated once the iframe is ready. Call `.init()` to kick off DOM insertion!
     this.iframePublicKey = null;
+
+    this.messageChannel = new MessageChannel();
   }
 
   /**
    * Inserts the iframe on the page and returns a promise resolving to the iframe's public key
    */
   async init(): Promise<string> {
     this.container.appendChild(this.iframe);
+    this.iframe.addEventListener("load", () => {
+      // Send a message to the iframe to initialize the message channel
+      this.iframe.contentWindow?.postMessage("init", this.iframeOrigin, [
+        this.messageChannel.port2,
```

Comment:
From the MDN docs: "The ownership of these objects [in the transfer array] is given to the destination side and they are no longer usable on the sending side".

So we're essentially giving the iframe the "port2" side of the pipe? Maybe that's worth a comment I'm not sure.

Also: should we do any feature detection at instantiation time to make sure MessageChannel is available?

Reply:
Added more comments and links to the MDN docs explaining exactly what's going on.